plugin/mlsdbonlinelocator.cpp

```diff
@@ -94,21 +104,20 @@ bool MlsdbOnlineLocator::findLocation(const QList<MlsdbProvider::CellPositioning
         return true;
     }
 
-    if (!haveFieldData(cells)) {
-        // no field data available
-        qCDebug(lcGeoclueMlsdbOnline) << "No field data available for MLS online request";
-        return false;
-    }
-
-    QNetworkRequest req(QUrl("https://location.services.mozilla.com/v1/geolocate?key=" + m_mlsKey));
-    req.setHeader(QNetworkRequest::ContentTypeHeader, "application/json");
-
     QVariantMap map;
-    map.unite(globalFields());
     map.unite(cellTowerFields(cells));
     map.unite(wifiAccessPointFields());
+    if (map.isEmpty() && !(m_fallbacksLacf || m_fallbacksIpf)) {
```

Comment:
if cell=0 wifi=0  m_fallbacksLacf=false m_fallbacksIpf=false, then we do not send a request to the server

Reply:
I would see this more readable with `map.isEmpty() && !m_fallbacksLacf && !m_fallbacksIpf`

Reply:
Ok. Changed